src/libstd/io/stdio.rs

```diff
@@ -674,20 +674,14 @@ fn print_to<T>(args: fmt::Arguments,
                local_s: &'static LocalKey<RefCell<Option<Box<Write+Send>>>>,
                global_s: fn() -> T,
                label: &str) where T: Write {
-    let result = match local_s.state() {
-        LocalKeyState::Uninitialized |
```

Comment:
This isn't quite the same behavior as before because `Uninitialized` would route to the global logger whereas this implementation would lazily run initialization and then route to the global logger.